services/web/client/source/resource/qxapp/node-meta-v1.0.0.json

```diff
@@ -0,0 +1,331 @@
+                        "examples": [
+                            "sun@sense.eight",
+                            "deleen@minbar.bab"
+                        ]
+                    },
+                    "affiliation": {
+                        "description": "Affiliation of the author",
+                        "type": "string",
+                        "examples": [
+                            "Sense8",
+                            "Babylon 5"
+                        ]
+                    }
+                }
+            }
+        },
+        "contact": {
+            "type": "string",
+            "format": "email",
+            "description": "email to correspond to the authors about the node",
+            "examples": [
+                "lab@net.flix"
+            ]
+        },
+        "inputs": {
+          "type": "object",
+          "description": "definition of the inputs of this node",
+          "patternProperties": {
+              "^[-_a-zA-Z0-9]+$": {
+                  "type": "object",
+                  "description": "all the input configurable for this service",
+                  "additionalProperties": false,
+                  "required": [
+                      "displayOrder",
+                      "label",
+                      "description",
+                      "type"
+                  ],
+                  "properties": {
+                      "displayOrder": {
+                          "type": "number",
+                          "description": "use this to numerically sort the properties for display",
+                          "examples": [
+                              1,
+                              -0.2
+                          ]
+                      },
+                      "label": {
+                          "type": "string",
+                          "description": "short name for the property",
+                          "examples": [
+                              "Age"
+                          ]
+                      },
+                      "description": {
+                          "type": "string",
+                          "description": "description of the property",
+                          "examples": [
+                              "Age in seconds since 1970"
+                          ]
+                      },
+                      "type": {
+                          "type": "string",
+                          "pattern": "^(number|integer|boolean|string|data:([^/\\s,]+/[^/\\s,]+|\\[[^/\\s,]+/[^/\\s,]+(,[^/\\s]+/[^/,\\s]+)*\\]))$",
+                          "description": "data type expected on this input glob matching for data type is allowed",
+                          "examples": [
+                              "number",
+                              "boolean",
+                              "data:*/*",
+                              "data:text/*",
+                              "data:[image/jpeg,image/png]",
```

Comment:
for now the idea is that the downstream nodes are accessing the data of the upstream nodes via an api provided by the upstream nodes ... the downstream nodes do not provide an api on their input side.

Reply:
I thought this was a way to connect the api to the downstream node.
output api port goes in input api port..

Reply:
well, the whole 'input' port part of the node description is 'in doubt' actually in an earlier version I had removed it completely, as I think this should probably not be handled at the node description level ... since the input mapping is too complex to be described as a static structure in general.